### PR TITLE
Adds indexing of items and topics

### LINKED ISSUE #39: Index Topics and Items
Enable the ability to index data from Topics and Items, and present the User with an interface to access these indexes. No specifics defined yet on fielded search, or other more involved interfaces, may not be needed. This issue covers the programmatic creation of the indexes. See issue #40 for UI component of search

### PR DESCRIPTION
**Pull request by richardrodgers**
closes #39

### CODE REVIEW

**Comment by richardrodgers** [comment]
@JPrevost : this PR wires in indexing of item and topic data when cataloging occurs (topic & item creation). The service URL will change (likely to $BONSAI_URL so we can run on HEROKU), but works for default local ElasticSearch. If you want to take a crack at the companion search piece (i.e UI, and the index lookup side), that would be great. You can examine my Topichib repo on Github, and the code in views/main.scala.html (which has the search box), and in controllers/Application#search (which has the code to call the search service) should provide a good start.

File: conf/application.conf [hunk 1]
```diff
@@ -67,6 +67,9 @@ logger.application=DEBUG
 # Email address of site administrator
 hub.admin.email="rrodgers@mit.edu"
 
+# URL of indexing service
+hub.index.url="http://localhost:9200/scoap3/"
```
Comment (JPrevost) [inline conf/application.conf]: Will this break on Heroku if we merge to master and it deploys as that won't exist? Maybe we need to switch to an ENV var for this before we merge? Or maybe the Indexer doesn't care if it fails at which point that's good for now (and horrible for later when we want to know it hasn't indexed something).

**Comment by richardrodgers** [comment]
Hmm, yea we probably should stub it with an ENV var, although I think ATM the Indexer will silently fail (not the best behavior). I'll look into setting up BONSAI_ES for our app on HEROKU. You can use the current application.conf value for local testing

**Comment by JPrevost** [comment]
I can confirm it fails silently so we can safely merge and maybe open a ticket to handle failed indexing in the future.

**Comment by JPrevost** [comment]
I think if we add a line after the localhost:9200 setting that redefines it to an env var url it will use the local and override with the ENV only if it exists... which might be perfect for this case.

**Comment by richardrodgers** [comment]
Yup, that's exactly the behavior we want. Can you merge this, then do a new PR for the addition to application.conf?

**Comment by JPrevost** [comment]
Mind if I just add a commit to this PR before we merge? Should only take a minute.

**Comment by richardrodgers** [comment]
Nope, that's fine too

**Comment by richardrodgers** [comment]
The ENV var name we want on Heroku is 'BONSAI_URL' - that plug-in is now activated

**Comment by richardrodgers** [comment]
Just thinking - this may not work as intended - the value of BONSAI_URL will be just the URL service root - it won't have the name of the 'index' (/scoap3/) which is  required part of the path. Can we do something like
hub.index.url=${?BONSAI_URL}+"/scoap3/"

?? If not we will have to alter the code/config, I think

**Comment by JPrevost** [comment]
We can change the ENV var in Heroku admin...

**Comment by richardrodgers** [comment]
I suppose, but as a platform supplied VAR, it makes me feel a little unclean...

**Comment by JPrevost** [comment]
I don't think we can just append the string as suggested as we'll be adding a string to an unset ENV var which will probably either fail with an error or set the url to "/scoap3/" if it is unset.

I think our options are to either set it as a ENV locally and not try to override or change the Heroku value.

**Comment by richardrodgers** [comment]
Let's go with changing Heroku app value for now (leave the PR as is)- I don't want to hold things up. What I was really trying to do is avoid hard-coding the index name in code, since I think there may be an obviously beneficial use-case for a multi-tenant ES server. But we can worry about all that much later....

**Comment by JPrevost** [comment]
:+1:

**Pull request merged**